tools/eslint-rules/snake-case-filenames.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview enforces snake_case for filenames.
+ * @author Shobhit Chittora <chittorashobhit@gmail.com>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const path = require('path');
+
+module.exports = function(context) {
+  const snakeCaseRegex = new RegExp('^[a-z_]+$');
+  const targetDirs = ['lib', 'deps', 'src'];
+
+  function checkIfInTargetDir(dirName) {
+    return targetDirs.filter((dir) => dirName.indexOf(dir) !== -1).length !== 0;
```

Comment:
Probably needs someone better versed in writing eslint rules (cc @nodejs/linting) than I am, but I think this check shouldn't be here and instead the rule should not be globally enabled for the project and only enabled on the desired subdirectories. 

The check as it stands would always return true if any part of the path had one of `targetDirs`, e.g., if your source tree was in something like `/home/user/src/node`.

This check is also one of the reasons the test is failing (since `camelCaseFilename.js` isn't in one of the `targetDirs`).

Reply:
Also I don't think there are any `.js` files in `src` so eslint won't run on any of those files?

Reply:
And it's probably not our remit to be requiring our upstream `deps` (e.g., `npm`) to be using snake case.